### PR TITLE
Add areAllVariationsSelected subject

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

Some stores need to display blocks conditionally based on if the user has selected all variations of a product. An example is `dzarm`, that wants to display the shipping simulator field only if the user has selected at least one product color.

#### How to test it?

https://kiwi--dzarm.myvtex.com/blusa-manga-curta-em-malha-de-viscose-6rykldken/p

Look for the shipping simulator after choosing a size variation.

#### Screenshots or example usage:

Using `condition.else`: 
image

Using `condition.product` with `areAllVariationsSelected is true`:
image
 

#### Describe alternatives you've considered, if any.

Adding this behavior specifically to the `shipping-simulator` component. However, this wouldn't solve the issue when an adjacent block needs to function in the same way, like the phrase accompanying the shipping simulator in the aforementioned page.

#### Related to / Depends on

https://vtex-dev.atlassian.net/jira/software/projects/SF/boards/105?selectedIssue=SF-385

### CODE REVIEW

**Review by klzns — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ If users interact with a product whose ID is not equal to 12, the block that is
 | `productClusters` | `array` | List of product clusters on the UI. |
 | `categoryTree` | `array` | List of categories on the UI. |
 | `specificationProperties` | `array` | List of product specifications. |
+| `areAllVariationsSelected` | `value` | `true` if every possible visible variations were selected, otherwise `false`. |
```
Comment (guerreirobeatriz) [inline docs/README.md:142]:
```suggestion
| `areAllVariationsSelected` | `value` | Whether all product variations available on the page were selected by the user (`true`) or not (`false`). | 
```

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

**Review by guerreirobeatriz — APPROVED** [review]

File: react/Condition.ts [hunk 2]
```diff
@@ -27,32 +27,29 @@ const Condition: StorefrontFunctionComponent<ConditionProps> = ({
...
   if (conditions == null) {
     // TODO: Handle error better
     console.warn('Missing conditions')
+
     return null
   }
 
-  if (!matches) {
+  // We use ctxMatch to prevent having both condition.xxx and condition.else blocks rendered at the same time.
+  if (!ctxMatch || !localMatch) {
```
Comment (kaisermann) [inline react/Condition.ts:52]: This is important 😁

Comment (klzns) [reply react/Condition.ts:52]: Any tests to get this scenario?

**Review by klzns — APPROVED** [review]

File: react/__tests__/Condition.test.tsx [hunk 3]
```diff
@@ -185,3 +186,38 @@ test('Switches from rendering the else component to a matched condition componen
   expect(queryByText(/Hooray!/)).toBeTruthy()
   expect(queryByText(/Oh no!/)).toBeFalsy()
 })
+
+test('Prevents Condition from rendering if context match is falsy', () => {
```
Comment (klzns) [inline react/__tests__/Condition.test.tsx:190]: I think the test is more about the behaviour than checking the internal structure

```suggestion
test('Prevents Condition from rendering at same time of Else', () => {
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — APPROVED** [review]
caprichou!

**Pull request merged**